### PR TITLE
[LDR R2] EOS-10513: NPE in S3 IAM operations

### PR DESCRIPTION
**Pull request by shalakadharap**

### CODE REVIEW

**Review by bkirunge7 — APPROVED** [review]

**Review by abhilekh268 — APPROVED** [review]

File: auth/server/src/main/java/com/seagates3/dao/ldap/AccessKeyImpl.java [hunk 1]
```diff
@@ -236,6 +236,7 @@ public AccessKey find(String accessKeyId) throws DataAccessException {
 
         AccessKeyStatus accessKeystatus;
         LDAPEntry entry;
+        if (ldapResults != null) {
         while (ldapResults.hasMore()) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/AccessKeyImpl.java:240]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/AccessKeyImpl.java [hunk 2]
```diff
@@ -303,6 +304,7 @@ public int getCount(String userId) throws DataAccessException {
              * TODO - Replace this iteration with existing getCount method if
              * available.
              */
+            if (ldapResults != null) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/AccessKeyImpl.java:307]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/AccountImpl.java [hunk 3]
```diff
@@ -254,6 +254,7 @@ public Account find(String name) throws DataAccessException {
             throw new DataAccessException("Failed to fetch accounts.\n" + ex);
         }
         int ldapResultCount = 0;
+        if (ldapResults != null) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/AccountImpl.java:257]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/RoleImpl.java [hunk 4]
```diff
@@ -142,7 +142,7 @@ public Role find(Account account, String roleName) throws DataAccessException {
                     + account.getName() + " pathPrefix: " + pathPrefix);
             throw new DataAccessException("Failed to find all roles.\n" + ex);
         }
-
+        if (ldapResults != null) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/RoleImpl.java:145]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/SAMLProviderImpl.java [hunk 5]
```diff
@@ -188,7 +188,7 @@ public SAMLProvider find(Account account, String providerName)
         } catch (LDAPException ex) {
             throw new DataAccessException("Failed to find IDPs.\n" + ex);
         }
-
+        if (ldapResults != null) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/SAMLProviderImpl.java:191]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/UserImpl.java [hunk 6]
```diff
@@ -265,6 +265,7 @@ public User findByUserId(String accountName, String userId)
                                           ex);
         }
         int ldapResultCount = 0;
+        if (ldapResults != null) {
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/UserImpl.java:268]: indentation

File: auth/server/src/main/java/com/seagates3/dao/ldap/UserImpl.java [hunk 7]
```diff
@@ -529,3 +530,4 @@ public void update(User user, String newUserName, String newPath)
     }
  }
 
+
```
Comment (amitwac2608) [inline auth/server/src/main/java/com/seagates3/dao/ldap/UserImpl.java:533]: you can avoid adding this newline.

**Review by amitwac2608 — COMMENTED** [review]
you can add or update existing UTs for this change.

**Pull request merged**